mitodl/odl-video-service
# add videos pagination backend

#### What are the relevant tickets?
#503 

#### What's this PR do?
1. Adds backend logic for videos pagination.

Note that there are no front-end changes yet.

#### How should this be manually tested?
1. Ensure that you have...
   a. a superuser with >= 3 videos
   b. a non-superuser with >= 3 videos
2. in `odl_video/settings.py`, find the line for `PAGE_SIZE_COLLECTIONS`, and make the default value be `2`.
3. As the superuser, go to http://localhost:8089/api/v0/videos/ . You should see all the videos, in paged format.
4. As the non-superuser, go to http://localhost:8089/api/v0/videos/ . You should see only videos for the non-superuser (or public videos, or videos which moira list criteria).
5. try different values for parameters `page_size=`, `page=` and `ordering=` (`title`, `-title`, `created_at`, `-created_at`) - the results should update accordingly.

#### Any background context you want to provide?
The meaty bit of this PR involved configuring/testing video permissions. However, this work will give us the infrastructure to do private videos (videos visible to only their owner), should we choose to pursue private videos.
